### PR TITLE
Get policy id endpoint

### PR DESCRIPTION
**Pull request by paweljakubas**
- [x] swagger update
- [x] type update
- [x] endpoint impl
- [x] unit test update
- [x] add integration testing

### Issue Number

adp-1191

### CODE REVIEW

File: lib/core-integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs [hunk 1]
```diff
@@ -3447,6 +3450,43 @@ spec = describe "NEW_SHELLEY_TRANSACTIONS" $ do
...
+    it "TRANS_NEW_CREATE_11a - Get policy id - incorrect template \
+        \" $ \ctx -> runResourceT $ do
+        wa <- fixtureWallet ctx
+        let payload = Json [json|{
+                "policy_script_template":
+                    { "all":
+                       [ { "active_from": 120 }
+                       ]
+                    },
+                "asset_name": "ab12"
```
Comment (piotr-iohk) [inline lib/core-integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs]: Hm, I'm quite sure that `asset_name` is not needed for getting a policy id... I believe that we need only `policy_script_template`.

Comment (paweljakubas) [reply lib/core-integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs]: indeed! I will revert. script template and policy public key are needed here! thanks

**Comment by paweljakubas** [comment]
bors try

**Review by piotr-iohk — APPROVED** [review]
LGTM. 
Added some e2e tests on top.

**Pull request merged**